sdk/clientcore/core/src/main/java/io/clientcore/core/http/pipeline/UserAgentOptions.java

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package io.clientcore.core.http.pipeline;
+
+/**
+ * The {@code UserAgentOptions} class is used to configure the user agent string for HTTP requests.
+ */
+public class UserAgentOptions {
+    private final String sdkName;
+    private final String sdkVersion;
+    private String applicationId;
+
+    /**
+     * Creates a {@link UserAgentOptions} with the specified SDK name and version.
+     *
+     * @param sdkName The name of the SDK.
+     * @param sdkVersion The version of the SDK.
+     */
+    public UserAgentOptions(String sdkName, String sdkVersion) {
+        this.sdkName = sdkName;
+        this.sdkVersion = sdkVersion;
+    }
```

Comment:
We have used `libraryName` and `libraryVersion` in [LibraryInstrumentationOptions](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/clientcore/core/src/main/java/io/clientcore/core/instrumentation/LibraryInstrumentationOptions.java#L56-L69). We should use the same names here as they refer to the same thing.

Reply:
another great catch, renamed `LibraryInstrumentationOptions` -> `SdkInstrumentationOptions` along with the properties